### PR TITLE
Adding abstract NPC and Obstacle interfaces

### PR DESCRIPTION
**Pull request by slynch0t**

### CODE REVIEW

File: GetUpChuckUnityProject/Assets/Scripts/Game World/Door.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Door : Obstacle, MonoBehaviour {
+
+	bool open = false;
+	
+	void OnCollisionEnter2D(Collision2D coll) {
+		if (coll.gameObject.tag == "Key") { condCheck(); }
```
Comment (AOjemann) [inline GetUpChuckUnityProject/Assets/Scripts/Game World/Door.cs]: use 
coll.gameObject.tag.Equals("Key")
for string comparison

condCheck needs to be passed an inventory

File: GetUpChuckUnityProject/Assets/Scripts/Game World/Pit.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Pit : Obstacle, MonoBehaviour {
+
+	bool inPit = false;
+	
+	void OnCollisionEnter2D(Collision2D coll) {
+		if (coll.gameObject.tag == "Player") { condCheck(); }
```
Comment (AOjemann) [inline GetUpChuckUnityProject/Assets/Scripts/Game World/Pit.cs]: use 
coll.gameObject.tag.Equals("Key")
for string comparison

condCheck needs to be passed an inventory

File: GetUpChuckUnityProject/Assets/Scripts/Game World/Plate.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Plate : Obstacle, MonoBehaviour {
+
+	public int weightCheck;
+	bool pressed = false;
+
+	void OnCollisionEnter2D(Collision2D coll) {
+		if (coll.gameObject.tag == "Player") { condCheck(); }
```
Comment (AOjemann) [inline GetUpChuckUnityProject/Assets/Scripts/Game World/Plate.cs]: use 
coll.gameObject.tag.Equals("Key")
for string comparison

condCheck needs to be passed an inventory

File: GetUpChuckUnityProject/Assets/Scripts/Game World/Tube.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Tube : Obstacle, MonoBehaviour {
+
+	public int weightCheck;
+	bool fits = false;
+
+	void OnCollisionEnter2D(Collision2D coll) {
+		if (coll.gameObject.tag == "Player") { condCheck(); }
```
Comment (AOjemann) [inline GetUpChuckUnityProject/Assets/Scripts/Game World/Tube.cs]: use 
coll.gameObject.tag.Equals("Key")
for string comparison

condCheck needs to be passed an inventory

**Comment by Murkantilism** [comment]
I can seem to find the change where condCheck is passed an inventory?

**Pull request merged**